modules/renter/hostdb/scan.go

```diff
@@ -136,10 +136,16 @@ func (hdb *HostDB) threadedProbeHosts() {
 	for hostEntry := range hdb.scanPool {
 		// Request settings from the queued host entry.
 		// TODO: use dialer.Cancel to shutdown quickly
-		hdb.log.Debugln("Scanning", hostEntry.NetAddress, hostEntry.PublicKey)
+		//
+		// A readlock is necessary when viewing the elements of the host entry.
+		hdb.mu.RLock()
+		netAddr := hostEntry.NetAddress
+		pubKey := hostEntry.PublicKey
+		hdb.mu.RUnlock()
```

Comment:
that's definitely weird... `scanPool` should probably send a `hostEntry` rather than a `*hostEntry` so you don't need to worry about it being modified. Why would an entry's netaddress or pubkey change though?